components/file_attachment.jsx

```diff
@@ -120,30 +120,46 @@ export default class FileAttachment extends React.PureComponent {
                         <div className='post-image__load'/>
                     )}
                 </a>
-                <div className='post-image__details'>
-                    <div
-                        className='post-image__detail_wrapper'
-                        onClick={this.onAttachmentClick}
-                    >
-                        <div className='post-image__detail'>
-                            <span className={'post-image__name'}>
-                                {trimmedFilename}
-                            </span>
-                            <span className='post-image__type'>{fileInfo.extension.toUpperCase()}</span>
-                            <span className='post-image__size'>{fileSizeToString(fileInfo.size)}</span>
-                        </div>
+            );
+
+            fileDetail = (
+                <div
+                    className='post-image__detail_wrapper'
+                    onClick={this.onAttachmentClick}
+                >
+                    <div className='post-image__detail'>
+                        <span className={'post-image__name'}>
+                            {trimmedFilename}
+                        </span>
+                        <span className='post-image__type'>{fileInfo.extension.toUpperCase()}</span>
+                        <span className='post-image__size'>{fileSizeToString(fileInfo.size)}</span>
                     </div>
-                    {canDownload &&
-                    <FilenameOverlay
-                        fileInfo={fileInfo}
-                        compactDisplay={compactDisplay}
-                        canDownload={canDownload}
-                        handleImageClick={this.onAttachmentClick}
-                        iconClass={'post-image__download'}
-                    >
-                        <DownloadIcon/>
-                    </FilenameOverlay>
-                    }
+                </div>
+            );
+        }
+
+        const canDownload = canDownloadFiles();
+        let filenameOverlay;
+        if (canDownload) {
+            filenameOverlay = (
+                <FilenameOverlay
+                    fileInfo={fileInfo}
+                    compactDisplay={compactDisplay}
+                    canDownload={canDownload}
+                    handleImageClick={this.onAttachmentClick}
+                    iconClass={'post-image__download'}
+                >
+                    <DownloadIcon/>
+                </FilenameOverlay>
+            );
+        }
+
+        return (
+            <div className='post-image__column'>
+                {fileThumbnail}
+                <div className='post-image__details'>
+                    {fileDetail}
+                    {filenameOverlay}
                 </div>
```

Comment:
Just dis some refactor here. Do not render elements which are not necessary for compact display.